knotx-knot/knotx-knot-fragment-splitter/src/test/java/io/knotx/splitter/impl/HtmlFragmentSplitterTest.java

```diff
@@ -102,18 +84,55 @@ public void testOneFragmentMiddle() throws Exception {
   }
 
   @Test
-  public void testOneFragmentEnd() throws Exception {
+  public void split_whenOneSnippetAtTheEnd_expectRawAndSnippetFragments() throws Exception {
+    String TEST_ONE_SNIPPET_END_HTML = "test-one-fragment-end.html";
+    List<Fragment> testOneSnippetEnd = new HtmlFragmentSplitter(DEFAULT_SCRIPT_TAG)
+        .split(FileReader.readText(TEST_ONE_SNIPPET_END_HTML));
     assertThat(testOneSnippetEnd.size(), equalTo(2));
     assertThat(testOneSnippetEnd.get(0).isRaw(), equalTo(true));
     assertThat(testOneSnippetEnd.get(1).isRaw(), equalTo(false));
     assertThat(testOneSnippetEnd.get(1).content(), equalTo(EXPECTED_ONE_FRAGMENT));
   }
 
   @Test
-  public void testOneFragmentWholeHtml() throws Exception {
+  public void split_whenOneSnippetIsWholeHtml_expectOneSnippetFragment() throws Exception {
+    String ONLY_ONE_SNIPPET_WHOLE_HTML = "test-one-fragment-whole.html";
+    List<Fragment> testOnlyOneSnippetWhole = new HtmlFragmentSplitter(DEFAULT_SCRIPT_TAG)
+        .split(FileReader.readText(ONLY_ONE_SNIPPET_WHOLE_HTML));
     assertThat(testOnlyOneSnippetWhole.size(), equalTo(1));
     assertThat(testOnlyOneSnippetWhole.get(0).isRaw(), equalTo(false));
     assertThat(testOnlyOneSnippetWhole.get(0).content(), equalTo(EXPECTED_ONE_FRAGMENT));
   }
 
+  @Test
+  public void split_whenCustomSnippetTagOneSnippet_expectThreeFragmentsAndSnippetInTheMiddle()
+      throws Exception {
+    String TEST_ONE_SNIPPET_MIDDLE_HTML = "test-custom-snippet-tag.html";
+    List<Fragment> testOneSnippetMiddle = new HtmlFragmentSplitter(CUSTOM_SCRIPT_TAG)
+        .split(FileReader.readText(TEST_ONE_SNIPPET_MIDDLE_HTML));
+    assertThat(testOneSnippetMiddle.size(), equalTo(3));
+    assertThat(testOneSnippetMiddle.get(0).isRaw(), equalTo(true));
+    assertThat(testOneSnippetMiddle.get(1).isRaw(), equalTo(false));
+    assertThat(testOneSnippetMiddle.get(1).content(), equalTo(
+        "<knotx:snippet data-knotx-knots=\"templating-X\" data-knotx-service=\"first-service\" type=\"text/knotx-snippet\"><h2>{{message}}</h2></knotx:snippet>"));
```

Comment:
if you are using scoped `<knot:x ... >` tag name ... why attributes have prefix `data-knotx`? :) I think in that case simplification will be nice 

`<knotx:snippet knots="" service="" ...>`

Reply:
Currently `<script`> is still the default tag, and `data-*` naming convention fits HTML5 naming rules.

Reply:
however `knotx:snippet` is not valid HTML5 tag ;)

Reply:
I mean.. that short named parameters to be available only in that new `knotx:snippet` so that you could keep it backward compatible